test/indexing_after_ns_removal.sh

```diff
@@ -64,13 +75,38 @@ done
 curl_es $es_pod /_cat/indices 2>&1 | artifact_out
 
 # Test indexing, if the amount of indices created is other than 20, exit with 1:
-amount_of_idx=$(oc rsh -c elasticsearch $espod es_util  --query=_cat/indices?h=index,docs.count | grep pelle | wc -l)
-if [ $amount_of_idx -ne 20 ]; then  exit 1; fi
+amount_of_idx=$(curl_es $es_pod /_cat/indices?h=index,docs.count | grep pelle | wc -l)
 
+if [ $amount_of_idx -ne 20 ]
+then  
+    os::log::error Amount of indices created is $amount_of_idx which is less than the amount of projects created
+    exit 1
+fi
 
 ## example of wrong indexing:
-# oc rsh -c elasticsearch $espod es_util  --query=_cat/indices?h=index,docs.count | grep pelle 
+# oc rsh -c elasticsearch $es_pod es_util  --query=_cat/indices?h=index,docs.count | grep pelle 
 
 # project.pelle-foo.53a8d3e3-42f7-11e8-a037-fa163e2c1fb0.2018.04.18                 3 
 # project.pelle-foo.5b71ac2c-42f7-11e8-a037-fa163e2c1fb0.2018.04.18                13 
 # project.pelle-foo.6b04cc03-42f7-11e8-a037-fa163e2c1fb0.2018.04.18                 4
+
+# verify that the kubernetes metadata in the record matches the object data
+# stored in temp_ns_uid and temp_pod_uid files
+# Note: If we reach this point, means there are 20 indices with one document
+# each, so no need to iterate through documents within an index but only over
+# the indices.
+
+for index in $(curl_es $es_pod /_cat/indices?h=index | grep pelle-foo)
+do
+    namespace_uid=$(curl_es $es_pod /"${index}"/_search?q=kubernetes.namespace_id:* | jq . | grep -i namespace_id | awk -F'"' '{ print $4 }')
+    grep -m 1 "${namespace_uid}" temp_ns_uid &>/dev/null
+    [ "$?" != "0" ] && os::log::error ${namespace_uid} does not match any UID collected from the actual namespace list during namespace creation
+
+    pod_uid=$(curl_es $es_pod /"${index}"/_search?q=kubernetes.namespace_id:* | jq . | grep -i pod_id | awk -F'"' '{ print $4 }')
+    grep -m 1 "${pod_uid}" temp_pod_uid &>/dev/null
+    [ "$?" != "0" ] && os::log::error ${pod_id} does not match any UID collected from the actual pod list during pod creation
+done
+
+
+# .orphaned index  is not created by default, so if it exists, then it contains records
+[[ $(curl_es $es_pod /_cat/indices?h=index,docs.count | grep .orphaned) ]] && os::log::error  There are records in the .orphaned index.
```

Comment:
Found one issue.  This last line returns 1 if .orphaned is not found.  That is, if successful, this test ends with (failure == 1); if .orphaned is found, (success == 0).
```
$ [[ $(cat /etc/hosts | grep .orphaned) ]] && echo "found"
$ echo $?
1
$ [[ $(cat /etc/hosts | grep localhost) ]] && echo "found"
found
$ echo $?
0
```
Can we do "if .orphaned is found, print error; exit 1; else exit 0"?

Reply:
@nhosoi Done, did some tests and throwing the expected result now.